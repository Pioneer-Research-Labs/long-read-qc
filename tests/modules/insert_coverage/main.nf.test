nextflow_process {

    name "Test Process insert_coverage"
    script "modules/insert_coverage/main.nf"
    process "insert_coverage"

    test("Should map insert coverage") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts.bam.bai'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/mapped_inserts_stats.tsv'),
                file("s3://pioneer-data/genomes/H_elongata/H_elongata_genes.gff".toString()),
                file("s3://pioneer-data/genomes/H_elongata/H_elongata_genes.bed".toString())
                )])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/insert_coverage.bed"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/insert_intersect.out"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/insert_coverage_full.bed"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/depth_report.tsv"),
            ("$outputDir/Gateway_EmptyBarcodedLib1_20241217/gene_coverage.bed")
            ).match()
        }

    }

}
