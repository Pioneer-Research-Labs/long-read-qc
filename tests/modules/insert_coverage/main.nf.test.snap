{
    "Should map insert coverage": {
        "content": [
            "insert_coverage.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
            "insert_intersect.out:md5,d41d8cd98f00b204e9800998ecf8427e",
            "insert_coverage_full.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
            "depth_report.tsv:md5,d41d8cd98f00b204e9800998ecf8427e",
            "gene_coverage.bed:md5,f151d2a1cf2ed3abf882a583bd2540a1"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T12:32:00.505133"
    }
}